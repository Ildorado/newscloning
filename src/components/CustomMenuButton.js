import React from 'react';
import {TouchableOpacity, View, Text, StyleSheet} from 'react-native';
import Colors from '../constants/Colors';
const CustomMenuButton = props => {
  return (
    <TouchableOpacity
      onPress={props.onPress}
      style={{...styles.MenuButton, ...props.style}}>
      <Text>{props.title}</Text>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  MenuButton: {
    backgroundColor: Colors.tertiary,
    height: 30,
    width: 80,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
    paddingHorizontal: 10,
    borderWidth: 1,
    borderColor: 'black',
  },
});
export default CustomMenuButton;
