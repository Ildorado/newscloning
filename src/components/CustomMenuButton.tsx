import React from 'react';
import {TouchableOpacity, StyleSheet, ViewStyle} from 'react-native';
import {Colors, WidthPoint} from '../constants/index';
import CustomText from '../constants/Styles/CustomText';
interface Props {
  onPress: () => void;
  title: string;
  style?: ViewStyle;
}
const CustomMenuButton: React.FC<Props> = ({onPress, title, style}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{...styles.MenuButton, ...style}}>
      <CustomText h3>{title}</CustomText>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  MenuButton: {
    backgroundColor: Colors.tertiary,
    height: 8 * WidthPoint,
    width: 22 * WidthPoint,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
    paddingHorizontal: 3 * WidthPoint,
    borderWidth: 1,
    borderColor: 'black',
  },
});
export default CustomMenuButton;
