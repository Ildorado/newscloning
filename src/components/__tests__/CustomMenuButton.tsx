import React from 'react';
import CustomMenuButton from '../CustomMenuButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <CustomMenuButton
        onPress={() => console.log('pressed')}
        title="title1"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
