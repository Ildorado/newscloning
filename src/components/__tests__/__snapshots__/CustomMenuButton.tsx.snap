// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EBE9EB",
      "borderColor": "black",
      "borderRadius": 10,
      "borderWidth": 1,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 22.5,
      "width": 165,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 28,
      }
    }
  >
    title1
  </Text>
</View>
`;
